src/components/collection/DownloadButton.js

```diff
@@ -0,0 +1,93 @@
+import GetAppIcon from '@material-ui/icons/GetApp';
+import IconButton from '@material-ui/core/IconButton';
+import React, { useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Tooltip,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Button,
+} from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import { MUTATION_KEYS } from '@graasp/query-client';
+import { QueryClientContext } from '../QueryClientContext';
+
+const DownloadButton = ({ id }) => {
+  const { t } = useTranslation();
+  const { useMutation } = useContext(QueryClientContext);
+  const {
+    mutate: downloadItem,
+    data,
+    isLoading,
+    isError,
+  } = useMutation(MUTATION_KEYS.DOWNLOAD_ITEM);
+
+  const [dialogOpen, setDialogOpen] = useState(false);
```

Comment:
What about a toast instead a dialog (so you don't need to worry about opening/closing as well as the button)? Check react-toastify and `notifier.js` where all trigger happens! Actually you can trigger the toast error directly from query-client -> I think by default you already created a routine, so you can reuse it in `notifier` ^^